### PR TITLE
Improve odr test

### PR DESCRIPTION
**Pull request by HDembinski**

### CODE REVIEW

**Comment by henryiii** [comment]
You may need to normalize paths on Windows in the Python script, I see slashes in both directions (`/` and `\`)

File: test/odr_test.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import os
+import sys
+import re
+
+this_path = os.path.dirname(__file__)
+
+all_headers = set()
+include_path = this_path + "/../include/"
```
Comment (henryiii) [inline test/odr_test.py]:
```suggestion
include_path = os.path.join(this_path, "..", "include")
```

File: test/odr_test.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+for root, dirs, files in os.walk(include_path):
+    for fn in files:
+        fn = root + "/" + fn
```
Comment (henryiii) [inline test/odr_test.py]:
```suggestion
        fn = os.path.join(root,  fn)
```

File: test/odr_test.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        assert fn.startswith(include_path)
```
Comment (henryiii) [inline test/odr_test.py]: I would probably make sure these are resolved with `os.path.abspath` first. Might be fine as is though.

File: test/odr_test.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    with open(filename) as f:
+        for hdr in re.findall('^#include [<"]([^>"]+)[>"]', f.read(), re.MULTILINE):
+            if not hdr.startswith("boost/histogram"):
+                continue
+            yield hdr
+
+
+included_headers = set()
+unread_headers = set()
+for hdr in get_headers(this_path + "/odr_test.cpp"):
```
Comment (henryiii) [inline test/odr_test.py]:
```suggestion
for hdr in get_headers(os.path.join(this_path, "/odr_test.cpp")):
```

This is why I love pathlib, plumbum.path, pretty much anything OO for paths!

File: test/odr_test.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    unread_headers.add(hdr)
+
+while unread_headers:
+    included_headers.update(unread_headers)
+    for hdr in tuple(unread_headers):
+        unread_headers.remove(hdr)
+        for hdr2 in get_headers(include_path + hdr):
```
Comment (henryiii) [inline test/odr_test.py]:
```suggestion
        for hdr2 in get_headers(os.path.join(include_path, hdr)):
```

**Comment by henryiii** [comment]
Instead of path.join everywhere, maybe do a sub of back slash for forward slash, probably on the walk. path.join will actualy put in the backslash, so the strings won't match what you pull out of the source unless you do abspath on them.

File: test/odr_test.py (hunk 1 above)
Comment (henryiii) [reply test/odr_test.py]: (These are suggestions to help debug/fix - I have not tested locally and do not have a Windows machine)

**Comment by henryiii** [comment]
I made a PR to your branch with a modified script that *should* work on Windows.

File: test/odr_test.py (hunk 1 above)
Comment (HDembinski) [reply test/odr_test.py]: Good point, will do. I just applied that fix to iminuit's setup.py as well the other day.

File: test/odr_test.py (hunk 3 above)
Comment (HDembinski) [reply test/odr_test.py]: What difference would abspath make?

File: test/odr_test.py (hunk 4 above)
Comment (HDembinski) [reply test/odr_test.py]: I don't mind using os.path.join for the time being.

**Comment by henryiii** [comment]
Have you looked at the version I provided in the PR? It is much simpler, and should have worked out of the box.

**Comment by HDembinski** [comment]
Already merged in 05c1bbfac12531e9f629ae40d1e53f47ffdbf1ee

**Comment by henryiii** [comment]
:+1:

**Pull request closed without merging**